src/codec/AuthDecoder.ts

```diff
@@ -0,0 +1,35 @@
+import ClientMessage = require("../ClientMessage");
```

Comment:
I think `AuthDecoder` and `AuthEncoder` classes will be replaced by `ClientAuthenticationCodec` in the future. So I suggest combining these two classes into one with name `ClientAuthenticationCodec` with methods `encodeRequest` and `decodeResponse` methods. So generated codec can fit very easily.